parallel-from-list/parallelFromList.groovy

```diff
@@ -0,0 +1,40 @@
+// While you can't use Groovy's .collect or similar methods currently, you can
+// still transform a list into a set of actual build steps to be executed in
+// parallel.
+
+// Our initial list of strings we want to echo in parallel
+def stringsToEcho = ["a", "b", "c", "d"]
+
+// The map we'll store the parallel steps in before executing them.
+def stepsForParallel = [:]
+
+// The standard 'for (String s: stringsToEcho)' syntax also doesn't work, so we
```

Comment:
The nice syntax works if you wrap the code in a function annotated as `@NonCps` (you should not use regular steps inside the non-cps function, which is actually feasible in this example).